socorro/external/postgresql/crash_migration_app.py

```diff
@@ -0,0 +1,100 @@
+database. Main use is for local development, to move the results of fakedata
+from PostgreSQL to another database.
+
+By default, it is configured to move crashes from PostgreSQL to Elasticsearch.
+'''
+
+from socorro.app.fetch_transform_save_app import main
+from socorro.collector.crashmover_app import RawAndProcessedCopierApp
+from socorro.external.crashstorage_base import CrashIDNotFound
+from socorro.external.postgresql.crashstorage import PostgreSQLCrashStorage
+from socorro.external.postgresql.dbapi2_util import (
+    SQLDidNotReturnSingleValue,
+    execute_query_iter,
+    single_value_sql,
+)
+
+
+class DumbPostgreSQLCrashStorage(PostgreSQLCrashStorage):
+    '''Dumb version of the PostgreSQL crashstorage class that fetches data
+    from the non-versioned tables. '''
+
+    def _get_raw_crash_transaction(self, connection, crash_id):
+        fetch_sql = 'select raw_crash from raw_crashes where uuid = %s'
+        try:
+            return single_value_sql(connection, fetch_sql, (crash_id,))
+        except SQLDidNotReturnSingleValue:
+            raise CrashIDNotFound(crash_id)
+
+    def _get_processed_crash_transaction(self, connection, crash_id):
+        fetch_sql = (
+            'select processed_crash from processed_crashes where uuid = %s'
+        )
+        try:
+            return single_value_sql(connection, fetch_sql, (crash_id,))
+        except SQLDidNotReturnSingleValue:
+            raise CrashIDNotFound(crash_id)
+
+
+class CrashMigrationApp(RawAndProcessedCopierApp):
+    app_name = 'crash_migration'
+    app_version = '1.0'
+    app_description = __doc__
+
+    @staticmethod
+    def get_application_defaults():
+        return {
+            'source.crashstorage_class': DumbPostgreSQLCrashStorage,
+            'destination.crashstorage_class':
+                'socorro.external.es.crashstorage.'
+                'ESCrashStorageNoStackwalkerOutput',
+        }
+
+    def source_iterator(self):
+        for x in self.get_all_crash_ids():
+            yield (x, {})  # (args, kwargs)
+        raise StopIteration()
+
+    def get_all_crash_ids(self):
+        connection = self.source.database.connection()
+        sql = 'select uuid from raw_crashes;'
+        return execute_query_iter(connection, sql)
+
+    def _transform(self, crash_id):
+        try:
+            raw_crash = self.source.get_raw_crash(crash_id)
+            processed_crash = self.source.get_unredacted_processed(
+                crash_id
+            )
+        except CrashIDNotFound:
+            self.config.logger.warning(
+                'cannot find crash %s',
+                crash_id,
+            )
+            return
+        except Exception:
+            self.config.logger.warning(
+                'error loading crash %s',
+                crash_id,
+                exc_info=True
+            )
+            return
+
+        if 'uuid' not in raw_crash:
+            raw_crash.uuid = crash_id
```

Comment:
Strange that that would ever happen. Isn't this based on `select uuid from raw_crashes`? And what is `raw_crash` here? A dict? A DotDict?